src/mount/linux.rs

```diff
@@ -161,3 +162,36 @@ pub fn umount2<P: ?Sized + NixPath>(target: &P, flags: MntFlags) -> Result<()> {
 
     Errno::result(res).map(drop)
 }
+
+libc_bitflags!(
+    /// Flags for [`open_tree()`].
+    pub struct OpenTreeFlags: c_uint {
+        /// Directly query the mount attached to the file descriptor.
+        AT_EMPTY_PATH as c_uint;
+        /// Do not trigger an automount target.
+        AT_NO_AUTOMOUNT as c_uint;
+        /// When cloning, also clone nested mount subtrees.
+        AT_RECURSIVE as c_uint;
+        /// If target is a symbolic link, do not dereference it.
+        AT_SYMLINK_NOFOLLOW as c_uint;
+        /// Clone the mount object.
+        OPEN_TREE_CLONE as c_uint;
+        /// Set the close-on-exec flag for the returned file descriptor.
+        OPEN_TREE_CLOEXEC as c_uint;
+    }
+);
+
+/// Find the mount object for the target path, and return it as a file-descriptor.
+///
+/// The returned FD behaves in the same way as those opened via `O_PATH`.
+pub fn open_tree<Fd: AsFd, P: ?Sized + NixPath>(
+    dirfd: Option<Fd>,
+    pathname: &P,
+    flags: OpenTreeFlags,
+) -> Result<OwnedFd> {
+    let res = pathname.with_nix_path(|cstr| unsafe {
+        let fd = dirfd.map(|v| v.as_fd().as_raw_fd()).unwrap_or(-1);
```

Comment:
We have a dedicated function for this conversion: [`at_rawfd()`](https://github.com/nix-rust/nix/blob/e7c877abf73f7f74e358f260683b70ce46db13b0/src/fcntl.rs#L433-L440), you can use it instead

Reply:
Ah thanks, I wasn't aware of that one, switched to it.